### PR TITLE
[ADD] api support

### PR DESCRIPTION
**Pull request by prakshalj0512**
- Checks the /health endpoint offered by the webserver for scheduler status
- Kills the airflow processes if hung

### CODE REVIEW

File: scheduler_failover_controller/failover/failover_controller.py [hunk 1]
```diff
@@ -160,35 +193,75 @@ def is_scheduler_running(self, host):
             # todo: If there's more then one scheduler running this should kill off the other schedulers. MIGHT ALREADY BE HANDLED. DOUBLE CHECK.
 
             is_running = active_list_length > 0
+            if (is_running):
+                url = "http://" + \
```
Comment (rssanders3) [inline scheduler_failover_controller/failover/failover_controller.py:197]: Can we confirm what version of Airflow this started coming with so we know what version of Airflow the ASFOC will work with?

**Review by rssanders3 — COMMENTED** [review]
Can wer revert the formatting changes as they arent needed and make it difficult to see what changes were actually made?

File: scheduler_failover_controller/failover/failover_controller.py [hunk 2]
```diff
@@ -160,35 +193,75 @@ def is_scheduler_running(self, host):
...
+                    str(host) + ":" + str(conf.get("webserver",
+                                                   "web_server_port")) + "/health"
+                try:
+                    response = requests.request("GET", url)
+                    json_data = response. json()
```
Comment (rssanders3) [inline scheduler_failover_controller/failover/failover_controller.py:202]: Remove space between `response.` and `json()`

File: scheduler_failover_controller/failover/failover_controller.py [hunk 3]
```diff
@@ -160,35 +193,75 @@ def is_scheduler_running(self, host):
...
+
+                    scheduler_status = json_data["scheduler"]["status"]
+                    if (str.lower(scheduler_status) == "healthy"):
+                        self.logger.info(
+                            "According to the webserver, is_running: " + str(is_running))
```
Comment (rssanders3) [inline scheduler_failover_controller/failover/failover_controller.py:207]: Incorrect log. Should be ""According to the webserver, scheduler_status: " + str(scheduler_status))"

File: scheduler_failover_controller/failover/failover_controller.py [hunk 4]
```diff
@@ -160,35 +193,75 @@ def is_scheduler_running(self, host):
...
+                        is_running = True
+
+                    else:
+                        is_running = False
+                        self.logger.info("Finished Checking if Scheduler on host '" + str(
```
Comment (rssanders3) [inline scheduler_failover_controller/failover/failover_controller.py:212]: Incorrect log. Should specify "scheduler_status: " + str(scheduler_status))"

File: scheduler_failover_controller/failover/failover_controller.py [hunk 5]
```diff
@@ -160,35 +193,75 @@ def is_scheduler_running(self, host):
...
+                            host) + "' is running. Seems the airflow scheduler is hung. According to the webserver, is_running: " + str(is_running))
+
+                        # Killing hung processes
+                        kill_scheduler_command = "ps -eaf | grep 'airflow scheduler' | awk '{print $2}' | xargs kill -9 $1"
```
Comment (rssanders3) [inline scheduler_failover_controller/failover/failover_controller.py:216]: Use the `shutdown_scheduler(self, host)` command for this.

**Review by rssanders3 — CHANGES_REQUESTED** [review]

**Comment by rssanders3** [comment]
Change the PR to push to the `development` branch

File: scheduler_failover_controller/failover/failover_controller.py (hunk 1 above)
Comment (prakshalj0512) [reply scheduler_failover_controller/failover/failover_controller.py:197]: The /health endpoint existed before but the info about the scheduler in the output was added in 1.10.2.

[https://issues.apache.org/jira/browse/AIRFLOW-3504](https://issues.apache.org/jira/browse/AIRFLOW-3504)

**Pull request closed without merging**